test/index.test.js

```diff
@@ -1003,6 +1055,30 @@ describe('Segment.io', function() {
             analytics.assert(res.fromDomain === 'userdata.example1.com');
             analytics.assert(!err);
           });
+
+          describe('isCrossDomainAnalyticsEnabled', function() {
+            it('should return false when crossDomainIdServers is null', function() {
+              segment.options.crossDomainIdServers = undefined;
+
+              assert.equal(segment.isCrossDomainAnalyticsEnabled(), false);
+            });
+
+            it('should return false when crossDomainIdServers is empty', function() {
+              segment.options.crossDomainIdServers = [];
+
+              assert.equal(segment.isCrossDomainAnalyticsEnabled(), false);
+            });
+
+            it('should return false when crossDomainIdServers is null', function() {
+              segment.options.crossDomainIdServers = [
+                'userdata.example1.com',
+                'xid.domain2.com',
+                'localhost'
+              ];
+
+              assert.equal(segment.isCrossDomainAnalyticsEnabled(), true);
+            });
+          });
```

Comment:
maybe a good idea to test for the case of crossDomainIdServers.length == 1, to catch off-by-1 error (even though one xid server doesn't really make sense, but we've seen this setting, and in this part of code XID should be considered enabled in that case)

Reply:
My thinking here was that the cases we care about is when it's 0 or not 0.

Reply:
(but I added a new test now for when `crossDomainIdServers.length == 1`